flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/DataTypeFactoryImpl.java

```diff
@@ -114,15 +111,18 @@ public <T> DataType createRawDataType(TypeInformation<T> typeInfo) {
     @Override
     public LogicalType createLogicalType(String typeString) {
         final LogicalType parsedType = LogicalTypeParser.parse(typeString, classLoader);
-        return parsedType.accept(resolver);
+        if (LogicalTypeChecks.hasNested(parsedType, t -> t.is(LogicalTypeRoot.UNRESOLVED))) {
```

Comment:
nit: I realise this is a backport. I wondered if we could add the typeString as an insert to the Exception message to help diagnostics. WDYT?

Reply:
Makes sense. Feel free to open a PR for this. This PR was a pure bugfix and focused on LogicalTypeDeduplicator which was just removed it at this location, without changing existing exception messages.

Reply:
@twalthr happy to do this, I wonder how I drive the exception in a realish unit test. I see you have added "NotInClassPathPojo" to the tests which I would think would drive the first unsupportedUserDefinedTypes() - but I don't see the table Exception being dealt with in the existing tests. Am I missing something?